### ISSUE TITLE
ERROR ITMS-90206: "Invalid Bundle. The bundle at 'MyApp.app/PlugIns/MyAppNotificationService.appex' contains disallowed file 'Frameworks'."

### ISSUE THREAD

**Issue by ndbeyer** [bitrise-io/bitrise#752]
- My app `MyApp` has a `MyAppNotificationService` extension.
- When building my app with `https://app.bitrise.io`, the `release.xcarchive` in the `bitrise artifacts tab` will contain a `frameworks` folder in `MyApp.app/PlugIns/MyAppNotificationService.appex` 
- The problematic `frameworks` folder contains standard swift libraries (see collapsible below)
- Therefore `Error: code -18000 (ERROR ITMS-90206: "Invalid Bundle. The bundle at 'MyApp.app/PlugIns/MyAppNotificationService.appex' contains disallowed file 'Frameworks'.")` during  `deploy-to-itunesconnect-application-loader` occurs
- **In contrast** When building `MyApp` locally with `bitrise cli` as [described here](https://discuss.bitrise.io/t/how-to-debug-your-build-locally-it-works-on-my-mac-pc-but-not-on-bitrise-io/392), the build won't contain the `frameworks` folder and `deploy-to-itunesconnect-application-loader` fullfills successfully
- The local build workflow with `bitrise-cli` skipped some steps in contrast to the one triggered on `https://app.bitrise.io`, which are
  - activate-ssh-key            
  - git-clone
  - cache-pull@2  
  - set-xcode-build-number 
  - certificate-and-profile-installer     
  - deploy-to-bitrise-io@1                
- In the next step i dropped the `Deploy to Bitrise.io - Apps, Logs, Artifacts` step from my `workflow` on `https://app.bitrise.io` and `deploy-to-itunesconnect-application-loader` fullfilled successfully

<details>
  <summary>Frameworks Folder</summary>

This is what is in `xcarchive/Products/Applications/MyApp.app/PlugIns/MyAppNotificationService.appex`

<img width="637" alt="grafik" src="https://user-images.githubusercontent.com/33200064/109017255-4a258e00-76b7-11eb-9705-8929b4605db6.png">  

</details>

In Summary, I think something is buggy with the `Deploy to Bitrise.io - Apps, Logs, Artifacts` step as it appears that this step created the `frameworks` folder that caused the error.

**Comment by bitce** [comment]
Hi @ndbeyer! Thank you for the detailed report! Can you please send us an URL for a build that generates this folder unnecessarily, so we can take a look at it?

**Comment by ndbeyer** [comment]
Thanks for your response @bitce, here is the url https://app.bitrise.io/build/aab4d81ae36979d8

**Comment by cathyharmon** [comment]
Hi @ndbeyer 

Can you please enable support access?  You can find this under the "Settings" tab on the app page.

Thanks!
cathy

**Comment by ndbeyer** [comment]
Hi @cathyharmon, thanks for your response - support was enabled.

**Comment by mjones-droid** [comment]
Hey @ndbeyer! I'm really sorry it took someone so long to get back to you on this. 

The thing that confuses me is that the Deploy to Bitrise.io step should only be taking the contents of certain directories (by default the $BITRISE_DEPLOY_DIR, which is /Users/vagrant/deploy) and uploading them to the App's & Artifacts tab. It's a "dumb" step in that it just takes whatever box you hand it and uploads it. No additional directories should be being created. 

It looks like when you had the Deploy to Bitrise step activated it uploaded the following files:

- /Users/vagrant/deploy/[REDACTED].dSYM.zip
- /Users/vagrant/deploy/[REDACTED].ipa
- /Users/vagrant/deploy/[REDACTED].xcarchive.zip
- /Users/vagrant/deploy/export_options.plist

I'm not even seeing a .app file in the deploy directory. 

Could you do me a favor and put the Deploy to Bitrise step back in, but at the very end of your workflow? I'd like to see if anything else gets added to the Apps & Artifacts tab as a result. 

I have a couple other thoughts on troubleshooting this before I file a bug report. Please send me a build once you've done the above and I'll continue digging into this.

**Comment by ndbeyer** [comment]
> Could you do me a favor and put the Deploy to Bitrise step back in, but at the very end of your workflow? I'd like to see if anything else gets added to the Apps & Artifacts tab as a result.

@mjones-droid I will try this
